lib/cocoapods-core/specification/dsl.rb

```diff
@@ -1324,6 +1324,42 @@ def subspec(name, &block)
         subspec
       end
 
+      # The test type this specification supports. This only applies to test specifications.
+      #
+      # ---
+      #
+      # @example
+      #
+      #   test_spec.test_type = :unit
+      #
+      # @param  [Symbol] type
+      #         The test type to use.
+      attribute :test_type,
```

Comment:
Instead the `Linter` takes care of validating the value is correct, which I added a test for.